src/pkcs15init/pkcs15-sc-hsm.c

```diff
@@ -282,6 +282,8 @@ static int sc_hsm_generate_key(struct sc_profile *profile, struct sc_pkcs15_card
 		r = SC_ERROR_NOT_IMPLEMENTED;
 		break;
 	}
+	if (r != SC_SUCCESS)
+		sc_pkcs15emu_sc_hsm_free_cvc(&cvc);
 	LOG_TEST_RET(p15card->card->ctx, r, "Could not encode GAKP cdata");
```

Comment:
The same here as pointed in the previous comments, it would be more understandable if the return was in the came block:
```suggestion
	if (r != SC_SUCCESS) {
		sc_pkcs15emu_sc_hsm_free_cvc(&cvc);
		LOG_TEST_RET(p15card->card->ctx, r, "Could not encode GAKP cdata");
	}
```